BraveRewardsUI/Pending Contributions/PendingContributionListController.swift

```diff
@@ -0,0 +1,225 @@
+    
+    contentView.tableView.register(PendingContributionCell.self)
+    
+    removeSelectedButton.isEnabled = false
+    
+    title = Strings.pendingContributionsTitle.capitalized
+    
+    navigationItem.rightBarButtonItem = editButton
+    toolbarItems = [
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
+      removeAllButton,
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    ]
+  }
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    navigationController?.setToolbarHidden(false, animated: animated)
+    reloadData()
+  }
+  
+  override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    navigationController?.setToolbarHidden(true, animated: animated)
+  }
+  
+  @objc private func tappedEdit() {
+    contentView.tableView.setEditing(true, animated: true)
+    navigationItem.setRightBarButton(doneButton, animated: true)
+    setToolbarItems([
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
+      removeSelectedButton,
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    ], animated: true)
+  }
+  
+  @objc private func tappedDone() {
+    contentView.tableView.setEditing(false, animated: true)
+    navigationItem.setRightBarButton(editButton, animated: true)
+    setToolbarItems([
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
+      removeAllButton,
+      .init(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    ], animated: true)
+  }
+  
+  // MARK: - Data
+  
+  private var contributionSections: [(type: RewardsType, contributions: [PendingContributionInfo])] = [] {
+    didSet {
+      removeAllButton.isEnabled = !contributionSections.isEmpty
+    }
+  }
+  
+  func reloadData() {
+    state.ledger.pendingContributions { [weak self] list in
+      guard let self = self else { return }
+      self.contributionSections = Dictionary(grouping: list, by: { $0.type })
+        .map { ($0.key, $0.value) }
+        .sorted(by: { $0.0.rawValue < $1.0.rawValue })
+      self.editButton.isEnabled = !list.isEmpty
+      self.contentView.tableView.reloadData()
+    }
+  }
+  
+  // MARK: - Actions
+  
+  @objc private func tappedRemoveAll(_ sender: UIBarButtonItem) {
+    contentView.tableView.performBatchUpdates({
+      self.contentView.tableView.deleteSections(IndexSet(0..<contributionSections.count), with: .automatic)
+      self.contributionSections.removeAll()
+    }, completion: { _ in
+      self.state.ledger.deleteAllPendingContributions { [weak self] _ in
+        self?.reloadData()
+      }
+    })
+  }
+  
+  @objc private func tappedRemoveSelected(_ sender: UIBarButtonItem) {
+    guard let selectedIndexes = contentView.tableView.indexPathsForSelectedRows else { return }
+    let contributionsToBeRemoved = selectedIndexes.map { contributionSections[$0.section].contributions[$0.row] }
```

Comment:
Not afraid of indexes going out of bounds here? Thinking if we should defend against it

Reply:
Not really since `indexPathsForSelectedRows` is going to match  `contributionSections` data (this table doesn't switch any data or do anything special, so its always going to be 1-1 data to index paths)